packages/scene-composer/src/components/panels/scene-settings/SceneTagSettingsEditor.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { Checkbox, FormField, Grid, Input } from '@awsui/components-react';
+
+import useLifecycleLogging from '../../../logger/react-logger/hooks/useLifecycleLogging';
+import { useStore } from '../../../store';
+import { sceneComposerIdContext } from '../../../common/sceneComposerIdContext';
+import { IComponentSettingsMap, ITagSettings, KnownComponentType, KnownSceneProperty } from '../../../interfaces';
+import { componentSettingsSelector } from '../../../utils/componentSettingsUtils';
+import { Slider } from '../Slider';
+
+export const SceneTagSettingsEditor: React.FC = () => {
+  useLifecycleLogging('SceneTagSettingsEditor');
+
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const intl = useIntl();
+  const setSceneProperty = useStore(sceneComposerId)((state) => state.setSceneProperty);
+  const getSceneProperty = useStore(sceneComposerId)((state) => state.getSceneProperty);
+  const tagSettings: ITagSettings = useStore(sceneComposerId)((state) =>
+    componentSettingsSelector(state, KnownComponentType.Tag),
+  );
+  const [dirty, setDirty] = useState(false);
+  const [focusInput, setFocusInput] = useState(false);
+  const [focusSlider, setFocusSlider] = useState(false);
+  const [draggingSlider, setDraggingSlider] = useState(false);
+
+  const [internalScale, setInternalScale] = useState(tagSettings.scale);
+
+  const updateSettings = useCallback(
+    (settingsPartial: Partial<ITagSettings>) => {
+      const newTagSettings: ITagSettings = {
+        ...tagSettings,
+        ...settingsPartial,
+      };
+      const newComponentSettings: IComponentSettingsMap = {
+        ...getSceneProperty(KnownSceneProperty.ComponentSettings),
+        [KnownComponentType.Tag]: newTagSettings,
+      };
+
+      setSceneProperty(KnownSceneProperty.ComponentSettings, newComponentSettings);
+    },
+    [tagSettings, getSceneProperty, setSceneProperty],
+  );
+
+  const onSliderChange = useCallback(
+    (event) => {
+      setInternalScale(Number(event.target.value));
+      setDirty(true);
+    },
+    [setInternalScale, setDirty],
+  );
+
+  const onInputBlur = useCallback(() => {
+    // If slider is getting focus, this makes sure to execute setFocusSlider(true) first to keep showing it.
+    const id = setTimeout(() => {
+      setFocusInput(false);
+    }, 1);
+    return () => {
+      clearTimeout(id);
+    };
+  }, [setFocusInput]);
+
+  const onInputChange = useCallback(
+    (event) => {
+      let value = event.detail.value;
+      if (value < 0) {
```

Comment:
Is there a max value we want to allow? And also what if they enter alphanumeric value?

Reply:
I don't see a need for max value since the tag will just look super big and won't break things

Reply:
only digits are allowed